lib/ansible/modules/cloud/amazon/aws_s3.py

```diff
@@ -582,7 +582,10 @@ def download_s3file(module, s3, bucket, obj, dest, retries, version=None):
 
     for x in range(0, retries + 1):
         try:
-            s3.download_file(bucket, obj, dest)
+            if version:
+                s3.download_file(bucket, obj, dest, ExtraArgs={'VersionId': version})
+            else:
+                s3.download_file(bucket, obj, dest)
```

Comment:
It'd be cleaner to have only kwargs conditional:
```suggestion
            optional_kwargs = {'ExtraArgs': {'VersionId': version}} if version else {}
            s3.download_file(bucket, obj, dest, **optional_kwargs)
```

Reply:
Oh.. `version` is immutable within the loop. You can move kwargs outside: https://github.com/ansible/ansible/pull/47867/files#r237441147
```suggestion
            s3.download_file(bucket, obj, dest, **optional_kwargs)
```